R/chunk.R

```diff
@@ -0,0 +1,23 @@
+#' Register Chunk Callback
+#'
+#' Registers a callback function to be executed after a specified chunk
+#'
+#' @param callback A callback function that returns a list of html output to be
```

Comment:
Should the callback function accept any arguments? Is there a way for callback functions to know if chunk execution failed or succeeded, or what happened after a chunk was executed?

This also seems a bit dangerous: since this API is available to anyone (not just Academy), this means that anyone using `rstudioapi` could inject arbitrary HTML after running a chunk.

We might need to talk about if there's a safer way for us to marshal HTML from Academy to RStudio. It sort of feels like we may want to keep the requisite UI code on the RStudio side, and use this callback to trigger and show that UI.

Reply:
The callback function accepts a `chunkName` and `chunkCode` with the code being optional. Is there a way to enforce this in R?

Reply:
You could check the function's formals (e.g. via `formals(callback)`) and validate that the function accepts at least two arguments (or accepts `...` to swallow arguments)

Reply:
That said, this validation is probably best performed on the RStudio side so no need to change things here -- just need to document the requirement.